backend/app/routers/authorization.py

```diff
@@ -18,23 +23,62 @@ async def save_authorization(
     user=Depends(keycloak_auth.get_current_username),
     db: MongoClient = Depends(dependencies.get_db),
 ):
-    """Save authorization info in Mongo. This is a triple of dataset_id/user_id/role."""
-    authorization_db = AuthorizationDB(**authorization_in.dict(), creator=user)
+    """Save authorization info in Mongo. This is a triple of dataset_id/user_id/role/group_id."""
+
+    # Check all connection and abort if any one of them is not available
+    if db is None:
+        raise HTTPException(status_code=503, detail="Service not available")
+        return
+
+    # Retrieve users from groups in mongo
+    user_ids: List[EmailStr] = []
+    group_q_list = db["groups"].find({"_id": {"$in": authorization_in.group_ids}})
```

Comment:
Do we have a mongo index on this field? Given this query we probably need one.

Reply:
Will create an index on group_ids